kratos/utilities/shifted_boundary_meshless_interface_utility.h

```diff
@@ -0,0 +1,363 @@
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "containers/pointer_vector.h"
+#include "includes/define.h"
+#include "includes/key_hash.h"
+#include "modified_shape_functions/modified_shape_functions.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @brief Utilities for the SBM-WTE extension operator calculation
+ * This class encapsulates several utilities for the calculation of the extension operator
+ * in the Shifted Boundary Method Without Taylor Expansions.
+ */
+class KRATOS_API(KRATOS_CORE) ShiftedBoundaryMeshlessInterfaceUtility
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    enum class ExtensionOperator
+    {
+        MLS,
+        RBF,
+        GradientBased
+    };
+
+    using IndexType = ModelPart::IndexType;
+
+    using NodeType = ModelPart::NodeType;
+
+    using GeometryType = ModelPart::GeometryType;
+
+    using ShapeFunctionsGradientsType = GeometryType::ShapeFunctionsGradientsType;
+
+    using ModifiedShapeFunctionsFactoryType = std::function<ModifiedShapeFunctions::UniquePointer(const GeometryType::Pointer, const Vector&)>;
+
+    using MeshlessShapeFunctionsFunctionType = std::function<void(const Matrix&, const array_1d<double,3>&, const double, Vector&)>;
+
+    using MLSShapeFunctionsAndGradientsFunctionType = std::function<void(const Matrix&, const array_1d<double,3>&, const double, Vector&, Matrix&)>;
+
+    using ElementSizeFunctionType = std::function<double(const GeometryType&)>;
+
+    using NodesCloudSetType = std::unordered_set<NodeType::Pointer, SharedPointerHasher<NodeType::Pointer>, SharedPointerComparator<NodeType::Pointer>>;
+
+    using CloudDataVectorType = DenseVector<std::pair<NodeType::Pointer, double>>;
+
+    using NodesCloudMapType = std::unordered_map<NodeType::Pointer, CloudDataVectorType, SharedPointerHasher<NodeType::Pointer>, SharedPointerComparator<NodeType::Pointer>>;
+
+    ///@}
+    ///@name Pointer Definitions
+
+    /// Pointer definition of ShiftedBoundaryMeshlessInterfaceUtility
+    KRATOS_CLASS_POINTER_DEFINITION(ShiftedBoundaryMeshlessInterfaceUtility);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// @brief Standard constructor
+    /// @param rModel Model container
+    /// @param ThisParameters Parameters object encapsulating the settings
+    ShiftedBoundaryMeshlessInterfaceUtility(
+        Model& rModel,
+        Parameters ThisParameters);
+
+    /// Copy constructor.
+    ShiftedBoundaryMeshlessInterfaceUtility(ShiftedBoundaryMeshlessInterfaceUtility const& rOther) = delete;
+
+    /// Destructor.
+    virtual ~ShiftedBoundaryMeshlessInterfaceUtility() = default;

```

Comment:
```suggestion
```

unless you actually derive from it I would remove it
The compiler does this job for you

Reply:
OK. Removing.